### PR TITLE
docs(table): update overview and code tabs content

### PR DESCRIPTION
**Pull request by Lenderson**
#### What is the purpose of this pull request?
update overview and code tabs content

#### How should this be manually tested?
Check the docs preview

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Quality improvement (tests or refactors)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Trivial change (small fix or feature that doesn't impact functionalities)
- [x] Requires change to documentation, which has been updated accordingly

### CODE REVIEW

File: packages/admin-ui-docs/content/table/code.mdx [hunk 1]
```diff
@@ -57,6 +56,14 @@ function Example() {
 
 </Section>
 
+<Section hasExplanation>
+
+## State
+
+Table is pretty much stateless. The state hook `useTableState` contains all business logic needed for the component.
```
Comment (marcelovicentegc) [inline packages/admin-ui-docs/content/table/code.mdx:63]:
```suggestion
The Table is stateless. The state hook `useTableState` contains all business logic needed for the component.
```

File: packages/admin-ui-docs/content/table/code.mdx [hunk 2]
```diff
@@ -383,7 +417,113 @@ function Example() {
...
+| type   | `plain`                                                   | Plain resolver type      | ✅       |
+| render | `(props: ResolverRenderProps<ReactNode, T>) => ReactNode` | Resolver render function | 🚫       |
+
+</Section>
+
+<Section hasExplanation>
+
+### Text
+
+The text resolver should be mainly used to render a text and a optional description just below it. It must be used for cells belonging to the name column
```
Comment (marcelovicentegc) [inline packages/admin-ui-docs/content/table/code.mdx:431]:
```suggestion
The text resolver should be mainly used to render a text and an optional description just below it. It must be used for cells belonging to the name column
```

**Review by marcelovicentegc — APPROVED** [review]
LGTM. Left two suggestions 😉

File: packages/admin-ui-docs/content/table/code.mdx (hunk 2 above)
Comment (anitavincent) [reply packages/admin-ui-docs/content/table/code.mdx:431]: Don't understand what you mean by "cells belonging to the name column"

Comment (anitavincent) [reply packages/admin-ui-docs/content/table/code.mdx:431]: Ah a column with type name? I think "belonging to a 'name' column" would be better if that's it

**Review by anitavincent — APPROVED** [review]

**Pull request merged**